src/Lib/Util/JWT.hs

```diff
@@ -55,10 +55,10 @@ decodeAndVerifyJWTToken :: (MonadIO m, MonadReader AppEnv m) => Text -> m (Maybe
 decodeAndVerifyJWTToken token = do
   secret <- JWT.secret <$> asks jwtSecret
   timeNow <- JWT.numericDate <$> liftIO getPOSIXTime
-  pure $ JWT.claims <$> JWT.decodeAndVerifySignature secret token >>= \claimsSet ->
-    case (,) <$> timeNow <*> JWT.exp claimsSet of
-      Just (now, expiryTimeStatedInToken) ->
-        if expiryTimeStatedInToken < now then
-          Nothing
-        else
-          jwtPayloadFromMap $ JWT.unregisteredClaims claimsSet
+  pure $ do
+    claimsSet <- JWT.claims <$> JWT.decodeAndVerifySignature secret token
+    (now, expiryTimeStatedInToken) <- (,) <$> timeNow <*> JWT.exp claimsSet
+    if expiryTimeStatedInToken < now then
```

Comment:
This `if` can be written slightly shorter using [`guard`](https://hackage.haskell.org/package/base-4.11.1.0/docs/Control-Monad.html#v:guard) function instead.